pygmt/src/filter1d.py

```diff
@@ -0,0 +1,144 @@
+"""
+filter1d - Time domain filtering of 1-D data tables
+"""
+import warnings
+
+import pandas as pd
+from pygmt.clib import Session
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import GMTTempFile, build_arg_string, fmt_docstring, use_alias
+
+
+@fmt_docstring
+@use_alias(
+    E="end",
+    F="filter",
```

Comment:
`filter` is a built-in function in Python, see https://docs.python.org/3/library/functions.html#filter. Sure we want to use this? Or use a parameter name like `filter_type` instead?

Reply:
Agree with @weiji14.

Reply:
That makes sense to me; I'll make the change